### PR TITLE
feat: add support for TypeScript 5.x

### PR DESCRIPTION
**Pull request by mrmckeb**
This PR adds support for TypeScript 5.x.

Once merged, this will be released as a beta for further testing.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -97,77 +117,140 @@ function init({ typescript: ts }: { typescript: typeof tsModule }) {
     // Create PostCSS processor.
     const processor = getProcessor(userPlugins);
 
-    // Create matchers using options object.
+    // // Create matchers using options object.
```
Comment (lianapache) [inline src/index.ts:120]:
```suggestion
    // Create matchers using options object.
```

**Review by lianapache — APPROVED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -20,14 +28,26 @@ const getPostCssConfigPlugins = (directory: string) => {
...
+
+    const languageServiceHostProxy = new Proxy(info.languageServiceHost, {
+      get(target, key: keyof LanguageServiceHost) {
+        return languageServiceHost[key]
+          ? languageServiceHost[key]
+          : target[key];
+      },
+    });
+
+    const languageService = createLanguageService(languageServiceHostProxy);
```
Comment (jakebailey) [inline src/index.ts:49]: Should this be using the ts parameter? If tsserver is run via VS Code, it may not match the version in node_modules.

Comment (jakebailey) [reply src/index.ts:49]: (Generally I'd suggest `type` importing only for plugins to avoid this mismatch)

Comment (jakebailey) [reply src/index.ts:49]: Also, you probably only want to import from tsserverlibrary, as it's a different bundle entirely.

**Pull request merged**